ado_asana_sync/sync/sync.py

```diff
@@ -408,35 +410,39 @@
                     f"{ado_task.fields['System.Title']}:found a matching asana task by name, updating task"
                 )
                 if asana_task is not None:
-                    item.asana_gid = asana_task.gid
+                    existing_match.asana_gid = asana_task.gid
                 update_asana_task(
                     a,
-                    item,
+                    existing_match,
                     tag,
                 )
                 continue
 
         # If already mapped, check if the item needs an update (ado rev is higher, or asana item is newer).
-        if item.is_current(a):
-            logging.info(f"{item.asana_title}:task is already up to date")
+        if existing_match.is_current(a):
+            logging.info(f"%s:task is already up to date", existing_match.asana_title)
             continue
 
         # Update the asana task, as it is not current.
-        logging.info(f"{item.asana_title}:task has been updated, updating task")
-        asana_task = get_asana_task(a, item.asana_gid)
+        logging.info(
+            f"%s:task has been updated, updating task", existing_match.asana_title
+        )
+        asana_task = get_asana_task(a, existing_match.asana_gid)
         if asana_task is None:
-            logging.error(f"No Asana task found with gid: {item.asana_gid}")
+            logging.error(f"No Asana task found with gid: %s", existing_match.asana_gid)
             continue
```

Comment:
**Suggestion:** Consider adding error handling for the case when 'get_asana_task' returns None. Currently, the code continues to the next iteration of the loop, but it might be beneficial to log an error or throw an exception in this case.
```suggestion
        asana_task = get_asana_task(a, existing_match.asana_gid)
        if asana_task is None:
            logging.error(f"No Asana task found with gid: %s", existing_match.asana_gid)
            raise ValueError(f"No Asana task found with gid: {existing_match.asana_gid}")
```